src/clock.rs

```diff
@@ -0,0 +1,156 @@
+#[cfg(not(feature = "std"))]
+use alloc::fmt;
+use embedded_time::duration::Milliseconds;
+pub use embedded_time::Clock;
+#[cfg(not(feature = "std"))]
+use libtock_drivers::result::FlexUnwrap;
+
+#[cfg(not(feature = "std"))]
+pub struct LibtockClock<const CLOCK_FREQUENCY: u32>(libtock_drivers::timer::Timer<'static>);
+#[cfg(not(feature = "std"))]
+impl<const CLOCK_FREQUENCY: u32> LibtockClock<CLOCK_FREQUENCY> {
+    pub fn new() -> Self {
+        let boxed_cb = alloc::boxed::Box::new(libtock_drivers::timer::with_callback(|_, _| {}));
+        let timer = alloc::boxed::Box::leak(boxed_cb).init().flex_unwrap();
+        Self(timer)
+    }
+}
+#[cfg(not(feature = "std"))]
+impl<const CLOCK_FREQUENCY: u32> fmt::Debug for LibtockClock<CLOCK_FREQUENCY> {
```

Comment:
In the list of PR we could send to embedded-time crate we should add this. The clock shouldn't need to implement Debug. This is a side effect of deriving Debug for embedded-time::Instant which adds that condition while actually what is needed is `Clock::T` to implement Debug.

The other PR was Fabian asking about missing `AddAssign` implementation for `Instant`.

Could you follow up on this? Either create issues to the crate (if they don't already exist) or implement the fix and send the PR.

Reply:
I'll follow up on this.